tests/subsys/tracing/prj.conf

```diff
@@ -0,0 +1 @@
+CONFIG_ZTEST=y
```

Comment:
what are we testing here exactly?

Reply:
Thanks @nashif ,  This is for the case when tracing functions disabled,   if tracing functions are disabled,  we shouldn't find any tracing data in the backend:
```
//checked in main.c
#ifdef CONFIG_TRACING
     zassert_true(thread_switched_found, "doesn't find thread_switched.");
     zassert_true(sys_trace_void_found, "doesn't find sys_trace_void.");
     zassert_true(sys_trace_semaphore_found,
             "doesn't find thread_idle_entered.");
#else
     /* Should produce any tracing data when tracing disabled. */
     zassert_false(thread_switched_found, "shouldn't have trace data.");
     zassert_false(sys_trace_void_found, "shouldn't have trace data.");
     zassert_false(sys_trace_semaphore_found, "shouldn't have trace data.");
#endif
```